### ISSUE TITLE
write* methods should throw if they have nothing to do

### ISSUE THREAD

**Issue by fischman** [jsxcad/JSxCAD#34]
Until #33 is implemented, passing an array of objects to writeThreejsPage silently emits nothing to the scene. It would have made debugging easier if it threw an exception complaining about nothing to do.

**Comment by pentacular** [comment]
I think that nothing to do probably shouldn't be an error.

Consider an operation like difference(x, x) -- this is kind of silly, but should legitimately produce an empty geometry.

More reasonably, within an assembly you may find that a component is sometimes completely displaced by subsequent components.

But in this case, when you passed an array of something rather than something, it should have complained about the data being malformed, rather than silently appearing to do nothing.

How about renaming the issue to something like "write methods should throw error on invalid input"?

**Comment by fischman** [comment]
Agreed they should throw on invalid input.
But also, I can't think of a reason you'd want an STL or .html file with no geometry in it, so I suspect that empty output is always operator error, so worth throwing for.
Up to you, obvs :)

**Comment by pentacular** [comment]
Yeah, let's add a flag into the options to enable error-on-empty.

We can default it true, and people can turn it off as necessary.

If it's sufficiently annoying we can always flip it to off by default.
